### PR TITLE
bin: swift support in update-ips

### LINKED ISSUE #1629: [1] Update-ips does not account for IPs for swift endpoint when using Thanos
**Describe the bug**:

As the title suggests, update-ips only does a look up of the IPs of the auth url.
It should also find the IPs of the swift endpoint.

E.g. elastx

- ops.elastx.cloud:5000, and
- swift.elastx.cloud:443

**To Reproduce**:

1. Set `thanos.objectStorage.type: swift`
2. Run update-ips and notice that it does not add the IPs/port for the Swift endpoint.

**Expected behaviour**:

Running update-ips adds all IPs nessecary for communicating applications to communicate with Swift.

**Version (add all relevant versions)**:

**Additional context**:

### PR DESCRIPTION
**Pull request by OlleLarsson**
**What this PR does / why we need it**:

bin/ck8s update-ips should now add the IPs and port for the swift endpoint.

**Which issue this PR fixes**: 
fixes #1629 

**Public facing documentation PR** *(if applicable)*

**Special notes for reviewer**:

I'll need to test this a bit more but so far on our support infra provider elastx it works just fine with using application credentials.

**Add a screenshot or an example to illustrate the proposed solution:**

**Checklist:**

- [x] Added relevant notes to [WIP-CHANGELOG.md](https://github.com/elastisys/compliantkubernetes-apps/blob/main/WIP-CHANGELOG.md)
- [x] Proper commit message prefix on all commits
- [ ] Updated the [public facing documentation](https://github.com/elastisys/compliantkubernetes)
- Is this changeset backwards compatible for existing clusters? Applying:
  - [x] is completely transparent, will not impact the workload in any way.
  - [ ] requires running a migration script.
  - [ ] will create noticeable cluster degradation.
        E.g. logs or metrics are not being collected or Kubernetes API server
        will not be responding while upgrading.
  - [ ] requires draining and/or replacing nodes.
  - [ ] will change any APIs.
        E.g. removes or changes any CK8S config options or Kubernetes APIs.
  - [ ] will break the cluster.
        I.e. full cluster migration is required.
- Chart checklist (pick exactly one):
  - [x] I upgraded no Chart.
  - [ ] I upgraded a Chart and determined that no migration steps are needed.
  - [ ] I upgraded a Chart and added [migration steps](https://github.com/elastisys/compliantkubernetes-apps/blob/main/migration).

**Pipeline config** *(if applicable)*
If you change some config options (e.g. add/rename variable or change the default value) you may need to update the config used by the pipeline in `pipeline/config`.

### CODE REVIEW

File: bin/update-ips.bash [hunk 1]
```diff
@@ -140,10 +151,27 @@ checkIfDiffAndUpdateKubectlIPs() {
...
+            updateIPs "${yaml_path}" "${file}" "${IPS[@]}"
+        else
+            log_warning "Diff found for ${yaml_path} in ${file//${CK8S_CONFIG_PATH}\//} (diff shows actions needed to be up to date)"
+        fi
+        has_diff=$(( has_diff + 1 ))
+    fi
+}
+
+# checkIfDiffAndUpdatePorts <yamlpath> <filepath> <ports>...
 checkIfDiffAndUpdatePorts() {
```
Comment (robinAwallace) [inline bin/update-ips.bash:172]: The other functions are flipped the other way around. Could you keep that?
So <something> <yamlpath>

File: bin/update-ips.bash [hunk 2]
```diff
@@ -184,6 +212,69 @@ yq_dig() {
...
+          }
+        }' "${auth_url}/auth/tokens")
+    fi
+
+    local os_token
+    local swift_url
+    os_token=$(echo "$response" | grep -oP "x-subject-token:\s+\K\S+")
+    swift_url=$(echo "$response" | tail -n +15 | jq -r '.[].catalog[] | select( .type == "object-store" and .name == "swift") | .endpoints[] | select(.interface == "public") | .url')
+
+    curl -i -s -X DELETE -H "X-Auth-Token: $os_token" -H "X-Subject-Token: $os_token" "${auth_url}/auth/tokens" > /dev/null
```
Comment (robinAwallace) [inline bin/update-ips.bash:273]: Why do you need to delete this token?

**Review by robinAwallace — APPROVED** [review]
Just some questions otherwise LGTM. Have not tested it tho.

File: bin/update-ips.bash (hunk 2 above)
Comment (OlleLarsson) [reply bin/update-ips.bash:273]: I think it's good not to have tokens laying around.
I think they are valid for 12 or 24 hours.

File: bin/update-ips.bash (hunk 1 above)
Comment (OlleLarsson) [reply bin/update-ips.bash:172]: I'm not sure If I want to change anything more, it was only the `checkIfDiffAndUpdatePorts` that had flipped the yaml_path and file arguments.
I think it is fine to have `<something> <yaml_path> <file> <something_else>`.

**Review by aarnq — APPROVED** [review]

**Comment by OlleLarsson** [comment]
Github is having issues with PRs, but I've tested this with password auth on Cleura public and compliant cloud, and password/app_cred on Elastx.
It seems to work just fine.

```
# Dry run on Cleura public cloud (don't think they mind having the IPs exposed)
# Cleura public cloud - password auth
--- sc-config.yaml
+++ expected
@@ -1 +1,4 @@
+- 91.106.196.168/32
+- 91.106.196.169/32
+- 91.106.196.170/32
 - 91.106.196.240/32
[ck8s] Diff found for .networkPolicies.global.objectStorageSwift.ips in sc-config.yaml (diff shows actions needed to be up to date)
--- sc-config.yaml
+++ expected
@@ -1 +1,2 @@
 - 5000
+- 8080
[ck8s] Diff found for .networkPolicies.global.objectStorageSwift.ports in sc-config.yaml (diff shows actions needed to be up to date)
```

File: bin/update-ips.bash (hunk 1 above)
Comment (OlleLarsson) [reply bin/update-ips.bash:178]: OK with this @robinAwallace ?

Comment (robinAwallace) [reply bin/update-ips.bash:178]: Yes this looks good :+1:

**Pull request merged**